x-pack/plugins/enterprise_search/server/lib/get_oauth_token_package_params.test.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ENTERPRISE_SEARCH_KIBANA_COOKIE } from '../../common/constants';
+
+import { getOAuthTokenPackageParams } from './get_oauth_token_package_params';
+
+describe('getOAuthTokenPackage', () => {
+  const tokenPackage = 'sEkReTZ';
+  const tokenPackageCookie = `${ENTERPRISE_SEARCH_KIBANA_COOKIE}=${tokenPackage}`;
+  const tokenPackageParams = { token_package: tokenPackage };
+
+  describe('when there are no cookie headers', () => {
+    it('returns an empty parameter set', () => {
+      expect(getOAuthTokenPackageParams(undefined)).toEqual({});
+    });
+  });
+
+  describe('when there is a single cookie header', () => {
+    it('returns an empty parameter set when our cookie is not there', () => {
+      const cookieHeader = '_st_fruit=banana';
+
+      expect(getOAuthTokenPackageParams(cookieHeader)).toEqual({});
+    });
+
+    it('returns the token package when our cookie is the only one', async () => {
```

Comment:
I don't think this needs to be async

```suggestion
    it('returns the token package when our cookie is the only one', () => {
```

Reply:
Oh cool! I was mostly parroting syntax I saw in other test files (where it seemed that every `it` was async). When *is* async required here?

Reply:
It's required if you have an `await` in your test/source code (usually an API call). This is a pretty static helper though so there isn't one AFAICT!